src/libasr/codegen/asr_to_python.cpp

```diff
@@ -619,6 +622,23 @@ class ASRToLpythonVisitor : public ASR::BaseVisitor<ASRToLpythonVisitor>
         s = r;
     }
 
+    void visit_ListConstant(const ASR::ListConstant_t &x) {
```

Comment:
Looking into your PRs https://github.com/lcompilers/lpython/pull/2690, https://github.com/lcompilers/lpython/pull/2691, https://github.com/lcompilers/lpython/pull/2692, I think the visitors implemented can have a common implementation using a templated function.

For example your `visit_List/Tuple/SetConstant()` can look like

```console
void visit_ListConstant(const ASR::ListConstant_t &x) {
    visit_AggregateConstant(x.n_args, x.m_args, "[", "]");
}

void visit_TupleConstant(const ASR::TupleConstant_t &x) {
    visit_AggregateConstant(x.n_elements, x.m_elements, "(", ")");
}

...
```

where `visit_AggregateConstant()` is the templated function. You can figure out the details for the arguments of this function.

Reply:
@Shaikh-Ubaid I totally agree with you. Except `DictConstant_t`, we can have a common visitor like you mentioned for the rest data-structures. :+1: